### PR TITLE
hotfix: Addresses Lint Warnings

### PR DESCRIPTION
**Pull request by subbramanil**
- Cleans up unused import statements
- Applies appropriate scope qualifiers (i.e., private) for variables & functions
- Moves hard-coded strings to String.xml & consts as required
- Cleans up Regular expressions by removing unnecessary character escapes

**Note:**
1. Almost all of the fixes were applied automatically through Android Studio's Auto-Suggestions which made sense & appropriate. None of the changes should affect the functionality of the app. No files were removed, although there were unused files & references.
2. Reports character escapes that are replaceable with the unescaped character without a change in meaning. Note that inside the square brackets of a character class, many escapes are unnecessary that would be necessary outside of a character class. For example the regex [\.] is identical to [.]
3. None of the changes should affect any functionality. But in case of any doubt, any change can be removed, analyzed & applied on another iteration.

Reference: #16

### CODE REVIEW

File: app/src/main/java/net/aiscope/gdd_app/ui/metadata/ListItem.kt [hunk 1]
```diff
@@ -3,6 +3,6 @@ package net.aiscope.gdd_app.ui.metadata
 data class ListItem(val id: Long, val name: String) {
 
     override fun toString(): String {
-        return name.orEmpty()
+        return name
```
Comment (eduardb) [inline app/src/main/java/net/aiscope/gdd_app/ui/metadata/ListItem.kt]: Haha, nice one

File: app/src/main/java/net/aiscope/gdd_app/ui/mask/customview/MaskCustomView.kt [hunk 2]
```diff
@@ -146,9 +147,7 @@ class MaskCustomView @JvmOverloads constructor(
             mY = y
 
             // TODO: improve performance, see if we can to canvas on touch-up
-            maskCanvas?.let {
-                it.drawPath(path, maskPaint)
-            }
+            maskCanvas?.drawPath(path, maskPaint)
```
Comment (eduardb) [inline app/src/main/java/net/aiscope/gdd_app/ui/mask/customview/MaskCustomView.kt]: 💯

File: app/src/main/res/layout/activity_new_health_facility.xml [hunk 3]
```diff
@@ -9,33 +9,35 @@
     <TextView
         android:id="@+id/text_health_facility_name_hint"
         style="@style/TextHint"
-        android:text="@string/text_health_facility_name_hint"
+        android:labelFor="@id/text_health_facility_name_field"
```
Comment (eduardb) [inline app/src/main/res/layout/activity_new_health_facility.xml]: I think that here the `android:text` should be kept as well, along with the new attribute.

Comment (subbramanil) [reply app/src/main/res/layout/activity_new_health_facility.xml]: Yes, You are correct.

File: app/src/main/res/layout/activity_new_health_facility.xml [hunk 4]
```diff
@@ -9,33 +9,35 @@
...
 
     <EditText
         android:id="@+id/text_health_facility_name_field"
         android:layout_width="0dp"
         android:layout_height="wrap_content"
         android:layout_marginTop="18dp"
         android:ems="10"
-        android:hint="Hospital name"
         android:inputType="textPersonName"
+        android:text="@string/text_health_facility_name_hint"
```
[the remark this answers is not in the document]
Comment (subbramanil) [reply app/src/main/res/layout/activity_new_health_facility.xml]: Yes, somehow I confused between the two attributes. :D

**Comment by subbramanil** [comment]
Updated the PR with the fixes.

**Pull request merged**